src/main/java/org/spongepowered/api/potion/PotionEffectType.java

```diff
@@ -25,10 +25,12 @@
 
 package org.spongepowered.api.potion;
 
+import org.spongepowered.api.attribute.AttributeSource;
+
 /**
  * Represents a possible type of {@link PotionEffect}.
  */
-public interface PotionEffectType {
+public interface PotionEffectType extends AttributeSource {
```

Comment:
If you drink a Speed Potion for example, MC will add a AttributeModifier to the Player which increases the Move Attribute. And remove it after the potion expires.
Iirc You could have look at MC's `Potion` class source code which is the equivalent in MC of `PotionEffectType` in SpongAPI.
Or maybe have a look at the issue in the description which gives little extra info.